integration/Jenkinsfiles/sonar.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     jcarsique
+ */
+
+ node('SLAVE') {
+    tool type: 'ant', name: 'ant-1.9'
+    tool type: 'hudson.model.JDK', name: 'java-8-oracle'
+    tool type: 'hudson.tasks.Maven$MavenInstallation', name: 'maven-3'
+    timeout(time: 5, unit: 'HOURS') {
+        timestamps {
+            stage('clone') {
+                checkout(
+                    [$class: 'GitSCM',
+                        branches: [[name: '*/${BRANCH}']],
+                        browser: [$class: 'GithubWeb', repoUrl: 'https://github.com/nuxeo/nuxeo'],
+                        doGenerateSubmoduleConfigurations: false,
+                        extensions: [
+                            [$class: 'WipeWorkspace'],
+                            [$class: 'CleanBeforeCheckout'],
+                            [$class: 'CloneOption', depth: 5, noTags: true, reference: '', shallow: true]
+                        ],
+                        submoduleCfg: [],
+                        userRemoteConfigs: [[url: 'git@github.com:nuxeo/nuxeo.git']]
+                    ])
+                sh """#!/bin/bash -xe
+                      ./clone.py $BRANCH -f $PARENT_BRANCH
+                """
+                shared = load("$WORKSPACE/integration/Jenkinsfiles/shared.groovy")
```

Comment:
Same as previously, use env.WORKSPACE would be safer.

Reply:
https://www.cloudbees.com/blog/top-10-best-practices-jenkins-pipeline-plugin
> 9. Don't: Set environment variables with env global variable

Contrary to the build parameters, it seems better to use environment variables as a copy.

Reply:
what I suggest is to get the workspace using env.WORKSPACE, not setting/modifying the env global 🤔